internal/kafka/constants/cluster.go

```diff
@@ -10,6 +10,9 @@ const (
 	// ClusterOperationDelete - OpenShift/k8s cluster delete operation
 	ClusterOperationDelete ClusterOperation = "delete"
 
+	// ClusterOperationHardDelete - OpenShift/k8s cluster hard delete operation
+	ClusterOperationHardDelete ClusterOperation = "hard_delete"
```

Comment:
Adding a new state makes me think we might need to update other places in the code where "deletion-related" states are considered / ignored.

Additionally, see my thoughts related to this in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1511#discussion_r1086722703